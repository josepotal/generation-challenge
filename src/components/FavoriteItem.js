import React from 'react';

const FavoriteItem = props => (
  <div className="fav-item">
    <p>#{props.store.id} | {props.store.Name}</p>
  </div>
);

export default FavoriteItem;
