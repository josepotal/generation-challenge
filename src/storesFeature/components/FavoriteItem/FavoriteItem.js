import React from 'react';
import PropTypes from 'prop-types';

import './FavoriteItem.module.scss';

const FavoriteItem = ({ store }) => (
  <div className="favItem">
    <p>
      #{store.id} | {store.Name}
    </p>
  </div>
);

FavoriteItem.propTypes = {
  store: PropTypes.shape({
    id: PropTypes.number.isRequired,
    Name: PropTypes.string.isRequired,
  }).isRequired,
};

export default FavoriteItem;
