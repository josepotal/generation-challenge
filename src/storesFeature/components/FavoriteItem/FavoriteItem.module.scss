@value colors: '../../../styles/variables.scss';
@value lightBlue from colors;

.favItem {
  border-bottom: 1px solid lightBlue;
  p {
    text-align: center;
    font-size: 0.7em;
  }
}
