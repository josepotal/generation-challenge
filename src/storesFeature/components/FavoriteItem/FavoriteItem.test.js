import React from 'react';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';

import FavoriteItem from './FavoriteItem';

describe('FavoriteItem', () => {
  let wrapper;
  let store;
  beforeEach(() => {
    store = {
      Address:
        'Valle  Santiago  N° 102  Col. Valle  de  Aragon  CP. 57100  Nezahualcoyotl  Estado  de  Mexico',
      Name: 'La Tienda Especiál 3777-Valle de Aragón ',
      id: 176,
      loc: [19.4893963, -99.05444829999999],
    };
    wrapper = shallow(<FavoriteItem store={store} />);
  });

  it('always renders a div', () => {
    expect(wrapper.find('div').length).toBe(1);
  });

  it('it should have favItem className', () => {
    expect(wrapper.find('.favItem').length).toBe(1);
  });

  it('it should receive a store prop', () => {
    expect(toJson(wrapper)).toMatchSnapshot();
  });
});
