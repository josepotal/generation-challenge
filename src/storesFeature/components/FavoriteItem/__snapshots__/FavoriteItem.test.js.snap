// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FavoriteItem it should receive a store prop 1`] = `
<div
  className="favItem"
>
  <p>
    #
    176
     | 
    La Tienda Especiál 3777-Valle de Aragón 
  </p>
</div>
`;
